frontend/simple/views/components/Transitions/MaskToModal/mixinTransitions.js

```diff
@@ -0,0 +1,167 @@
+      this.updateSpecsOf(targetInner, 'target')
+
+      // Fade out the targetInner after the user interaction...
+      Velocity(targetInner, { opacity: 0 }, { duration: this.config.fadeBack / 2, delay: this.config.fadeBack / 2 })
+
+      // But only fades out complety the Target (that has the dark background)
+      // after Masker goes back to the initial position
+      Velocity(el, { opacity: 0 }, { duration: this.config.fade, delay: this.config.maskerTime, complete })
+    },
+    elementStartsInvisible (el) {
+      // Use only opacity to fadeIn/Out because it's faster than
+      // Velocity's 'fadeIn' feature (it doesn't use display)
+      Velocity(el, { opacity: 0 }, { duration: 0 })
+    },
+
+    // -- Masker animations
+    // when animates to Target
+    maskEnter (el, complete) {
+      console.log('maskLeave')
+
+      this.maskerTakesImmediatelyTheShapeOf(el, 'trigger', this.config.fade)
+      this.maskerAnimatesToTheShapeOf(el, 'target', complete)
+    },
+
+    // when animates back to a Trigger
+    maskLeave (el, complete) {
+      console.log('maskEnter')
+
+      this.maskerTakesImmediatelyTheShapeOf(el, 'target', this.config.fadeBack)
+      this.maskerAnimatesToTheShapeOf(el, 'trigger', complete)
+    },
+
+    // -- Animation utils
+    maskerTakesImmediatelyTheShapeOf (maskerEl, originalElement, duration) {
+      // Masker takes imediately the specs (dimensions and position)
+      // of the originalElement (Trigger or Target) size...
+      Velocity(maskerEl, { opacity: 0, ...this.getSpecsOf(originalElement) }, { duration: 0 })
+
+      // And fadesIn at the same time as the originalElement (trigger or target) is fading Out
+      Velocity(maskerEl, { opacity: 1 }, { duration })
+    },
+    maskerAnimatesToTheShapeOf (el, originalElement, complete) {
+      // Then it animates to the the originalElement specs creating the "growing effect"
+      Velocity(el, { ...this.getSpecsOf(originalElement) }, { duration: this.config.maskerTime, easing: 'ease-out' })
+      // And finally it fades out at the same time Target is fading in.
+      Velocity(el, { opacity: 0 }, { duration: this.config.fade, complete })
+    },
+    updateSpecsOf (el, elementId) {
+      this.MaskToModal.updateSpecsOf(el, elementId)
+    },
+    getSpecsOf (elementId) {
+      return this.MaskToModal.elementsSpecs[elementId]
+    },
+    getTargetInnerWhen (scenario) {
+      // Why targetInner is .modal-card:
+      // We want the whole Target (modal) to fade, (modal-card + modal-background)
+      // but we want the Masker to take just the .modal-card shape instead
+
+      // - REVIEW: If you know a easier way to get the modal's .card element,
+      // please let me know how to do it!
+      if (scenario === 'enter') {
+        // this.targetCard is not available yet,
+        // so we need to access by using $slots.default
+        return this.$slots.default[0].componentInstance.$refs.modal.$refs.card
```

Comment:
It's not clear to me what this `componentInstance` is. Where did it come from? I can't find it in Vue.js docs either.